src/zks_wallet/wallet.rs

```diff
@@ -553,12 +548,17 @@ where
             )
             .await?;
 
-        era_provider
+        let tx_receipt = era_provider
             .get_transaction_receipt(response.1)
             .await?
             .ok_or(ZKSWalletError::CustomError(
                 "No transaction receipt for withdraw".to_owned(),
-            ))
+            ))?;
+
+        era_provider
+            .wait_for_finalize(tx_receipt, None, None)
+            .await
+            .map_err(|error| ZKSWalletError::CustomError(error.to_string()))
```

Comment:
We could try this here since we implement `ZKSWalletError::ProviderError` from ethers's `ProviderError`
```suggestion
        era_provider
            .wait_for_finalize(tx_receipt, None, None)
            .await?
```